packages/manager/apps/telecom/src/app/telecom/pack/resiliation/pack-resiliation.controller.js

```diff
@@ -4,200 +4,215 @@ import remove from 'lodash/remove';
+                `pack_resiliation_choice_${reason}`,
+              ),
+            }));
         }
 
-        if (self.resiliationTerms.name) {
-          self.resiliationTerms.nameToDisplay = self.resiliationTerms.name;
+        if (this.resiliationTerms.name) {
+          this.resiliationTerms.nameToDisplay = this.resiliationTerms.name;
         } else {
-          self.resiliationTerms.nameToDisplay = self.resiliationTerms.packName;
+          this.resiliationTerms.nameToDisplay = this.resiliationTerms.packName;
         }
 
-        self.updateFeeSummary();
+        this.updateFeeSummary();
       })
-      .catch((err) => new TucToastError(err))
+      .catch((err) => new this.TucToastError(err))
       .finally(() => {
-        self.loading = false;
+        this.loading = false;
       });
 
-    self.subServicesTermsLoading = true;
-    self.subServicesTerms = null;
-    self.subServicesTermsError = false;
-    /**
-     * Validator for the resiliationDate
-     * @param {Date} specifiedDate Date to validate
-     */
-    this.checkDate = function checkDate() {
-      return (
-        !self.model.when ||
-        moment(self.model.when).isSame(self.minResiliationDate) ||
-        moment(self.model.when).isAfter(self.minResiliationDate)
-      );
-    };
-
-    OvhApiPackXdslResiliation.Aapi()
+    this.subServicesTermsLoading = true;
+    this.subServicesTerms = null;
+    this.subServicesTermsError = false;
+    this.OvhApiPackXdslResiliation.Aapi()
       .subServicesTerms({
-        packId: $stateParams.packName,
+        packId: this.$stateParams.packName,
       })
       .$promise.then((data) => {
-        self.subServicesTerms = data;
+        this.subServicesTerms = data;
       })
       .catch((err) => {
-        self.subServicesTermsError = true;
-        return new TucToastError(err);
+        this.subServicesTermsError = true;
+        return new this.TucToastError(err);
       })
       .finally(() => {
-        self.subServicesTermsLoading = false;
+        this.subServicesTermsLoading = false;
       });
-  };
-
-  /**
-   * Get the current currency symbol
-   *
-   * @return promise with the symbol of the current currency
-   */
-  this.getCurrentCurrencySymbol = function getCurrentCurrencySymbol() {
-    return $q
-      .when(coreConfig.getUser())
-      .then((me) => (me && me.currency ? me.currency.symbol : ''));
-  };
+  }
 
   /**
    * Validator for the resiliationDate
    * @param {Date} specifiedDate Date to validate
    */
-  this.checkDate = function checkDate() {
+  checkDate() {
     return (
-      !self.model.when ||
-      moment(self.model.when).isAfter(self.minResiliationDate)
+      !this.model.when ||
+      moment(this.model.when).isSame(this.minResiliationDate) ||
+      moment(this.model.when).isAfter(this.minResiliationDate)
     );
-  };
+  }
+
+  /**
+   * Get the current currency symbol
+   *
+   * @return promise with the symbol of the current currency
+   */
+  getCurrentCurrencySymbol() {
```

Comment:
**question:** This method still need to be asynchronous?

Reply:
Yes